extensions/wikia/CreateNewWiki/js/CreateNewWiki.js

```diff
@@ -69,32 +69,18 @@
 						wikiDomain: self.wikiDomain.val(),
 						wikiLang: self.wikiLanguage.find('option:selected').val()
 					});
-					if (self.$authWrapper.length) {
-						// Init user auth
-						self.userAuth = {
-							el: self.$authWrapper,
-							loginAjaxForm: new UserLoginAjaxForm(
-								'#UserAuth .UserLoginModal',
-								{
-									ajaxLogin: true,
-									callback: function () {
-										self.transition('UserAuth', true, '+');
-									}
-								}
-							)
-						};
-						if (typeof FacebookLogin !== 'undefined') {
-							FacebookLogin.callbacks['login-success'] = function () {
-								self.transition('UserAuth', true, '+');
-								FacebookLogin.closeSignupModal();
-							};
-						}
+					if (window.wgUserName) {
+						self.transition('NameWiki', true, '+');
+					} else {
+						var redirectUrl = encodeURIComponent(window.location.href + '&wikiName=' + self.wikiName.val() + '&wikiDomain=' + self.wikiDomain.val() + '&wikiLanguage=' + self.wikiLanguage.find('option:selected').val());
```

Comment:
We perform very similar calculation in `$signupRedirect.submit` callback. What do you think about extracting it?